tests/unit_tests/crud/test_missing_attributes.py

```diff
@@ -0,0 +1,96 @@
+import uuid
+
+from elasticsearch_dsl import AttrDict
+
+from app.api.collections.missing_attributes import (
+    missing_attributes_search,
+    missing_attributes_spec,
+    missingPropertyFilter,
+)
+from app.api.collections.models import MissingMaterials
+from app.api.collections.utils import map_elastic_response_to_model
+
+
+def test_missing_attributes_search():
+    dummy_uuid = uuid.uuid4()
+    dummy_attribute = "properties.cm:title"
+    dummy_missing_attribute = missingPropertyFilter[0].value
+    print(dummy_missing_attribute)
+    dummy_maximum_size = 3
+    search = missing_attributes_search(
+        dummy_uuid, dummy_missing_attribute, dummy_maximum_size
+    )
+    assert search.to_dict() == {
```

Comment:
Eventually it makes sense to just kick out the whole elastic dsl depenency...

I mean, there is tons framework / utility code to build a hand full of different queries, and then there is tests that test that framework code, where we could just build the queries exactly like demonstrated in that test here.

That will make stuff way more consolidated and maintainable, eventually on the cost of a little repetition... It's a tradeof :thinking:

Reply:
I agree partially, a lot of unnecessary code is built around elastic - but mostly by the former maintainer. If we want to get rid of elasticsearch_dsl, we would have to drop the complete Search() etc. stuff, i.e., implement our own algorithm to talk with Elasticsearch. Currently, I see no need to do just that. But reducing complexity is high on my list here.